Client/Frontend/Home/BookmarksEditPanel.swift

```diff
@@ -0,0 +1,335 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import Storage
+import Shared
+import XCGLogger
+
+private let log = Logger.browserLogger
+
+enum DetailPanel: Int {
+    case Title = 0
+    case URL
+    case Folder
+    case _numberOfPanels
+}
+
+struct BookmarksEditPanelUX {
+    private static let LabelLeftRightPadding = 20
+    private static let LabelTopBottomPadding = 10
+    private static let TextFieldTopBottomPadding = 5
+}
+
+class BookmarksEditPanel: SiteTableViewController, HomePanel, UITextFieldDelegate {
```

Comment:
Nit: Completely a personal preference but instead of listing the various protocols this view controller conforms to here, I like to break each one out (when it makes sense) and add them as `extension`s to the class. I find it makes the code cleaner, more readable, and encourages thinking about protocols as 'aspects' or 'traits' of a class.